### PR TITLE
mavlink: don't reset _src_addr_initialized on parameter updates

### PR DESCRIPTION
**Pull request by dagar**
https://github.com/PX4/PX4-Autopilot/issues/19215

### CODE REVIEW

File: src/modules/mavlink/mavlink_main.cpp [hunk 1]
```diff
@@ -2229,6 +2229,14 @@ Mavlink::task_main(int argc, char *argv[])
...
 	}
 
+#if defined(CONFIG_NET)
+
+	if (!multicast_enabled()) {
+		_src_addr_initialized = false;
+	}
+
+#endif // CONFIG_NET
+
```
Comment (lukegluke) [inline src/modules/mavlink/mavlink_main.cpp:2239]: Is it really needed here? _src_addr_initialized is false by default and will be set to true upper if there is multicast option.

Comment (dagar) [reply src/modules/mavlink/mavlink_main.cpp:2239]: I didn't think so, but I haven't dug into the original intent of the change. 

Let's try getting rid of it.

**Comment by dagar** [comment]
This made a bit more sense when MAV_BROADCAST was a single parameter accessed directly. Unnecessarily clearing `_src_addr_initialized` was still questionable, but at least the history makes more sense now.
Screenshot from 2022-02-18 09-49-54

We can safely remove this entirely.

**Comment by lukegluke** [comment]
> We can safely remove this entirely.

Agree. Anyway I've tested this just in case, works ok.

**Comment by dagar** [comment]
> @dagar by the way I don't see MAV_X_BROADCAST parameters in [docs](https://docs.px4.io/master/en/advanced_config/parameter_reference.html) for master. Aren't they autogenerated?

It's generated per mavlink instance here. https://github.com/PX4/PX4-Autopilot/blob/d5d50d585592b43f1e0db38c10f248411b144371/src/modules/mavlink/module.yaml#L141-L155

A lot of the parameter generation is now based on the specific board, which is slightly problematic for the generated docs in a few cases.

Probably the best fix in this case would be to update SITL so that it's network is configured just like the NuttX boards with the Mavlink UDP instance parameter configurable. The slightly harder part to work through is the existing SITL mavlink network startup where there's special handling of different port numbers per SITL instance. https://github.com/PX4/PX4-Autopilot/blob/master/ROMFS/px4fmu_common/init.d-posix/px4-rc.mavlink

**Pull request merged**